### ISSUE TITLE
PR-analysis on "new" code not working as expected

### ISSUE THREAD

**Issue by Qwby** [mc1arke/sonarqube-community-branch-plugin#161]
**Situation**
I'm trying to use the plugin in order to have a quality gate for pull requests that basically says "new lines of code in this PR needs to be x% covered by tests" or "x issues in new code" with the community version. I'm not sure why, but SonarQube doesn't understand "new code" in a PR analysis should mean, code that's new with this pull request. It can't be comparing the pull request against the base branch "develop", like it says, because there are a ton of "new" issues when the PR contains no code changes at all. It must be comparing it against master or some older software version.

I am unsure if the problem is to be found with this plugin or if I just didn't configure something right.

**Expected behavior**
There should be no issues on new code with a PR that has no code changes.

**Example Call from Jenkins**
mvn sonar:sonar -Dsonar.pullrequest.key=13 -Dsonar.pullrequest.branch=feature/KEY-12-some-feature -Dsonar.pullrequest.base=develop

**Software Versions**
 - SonarQube Version: 7.9.1
 - Plugin Version: 1.3.0
 - Sonar-Maven-Plugin Version: 3.7.0.1746

If you need more information from me, just let me know.

**Comment by mc1arke** [comment]
Do you have a project I can run analysis against where I can replicate this? Part of my testing of the release candidates of this plugin is to make sure it only picks up new issues so your evidence doesn't fit what I'd expect.

**Comment by Qwby** [comment]
Looking back I was probably too quick to raise an issue about it, I figured it out - somehow I had no idea on Friday what would caused this, but today I found that I didn't execute the maven profile I thought I had executed. The right maven profile had all the correct exclusions for SonarQube, so there were a ton of "new issues" because of that.

For anyone else having this problem in the future:
- check if your Sonar exclusions have changed
- check if your configuration of the Sonar Scanner Plugin inside of your pom makes sense
- check if you're actually executing the right Maven profile
- looking at where the "new" issues were found might give you a hint on what's wrong

I'm closing this, since this wasn't an issue with the plugin.
